java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertTextBlockToString.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.hints.jdk;
+
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.util.TreePath;
+import org.netbeans.api.java.lexer.JavaTokenId;
+import org.netbeans.api.java.source.CompilationInfo;
+import org.netbeans.api.lexer.TokenSequence;
+import org.netbeans.spi.editor.hints.ErrorDescription;
+import org.netbeans.spi.editor.hints.Fix;
+import org.netbeans.spi.java.hints.ErrorDescriptionFactory;
+import org.netbeans.spi.java.hints.Hint;
+import org.netbeans.spi.java.hints.HintContext;
+import org.netbeans.spi.java.hints.JavaFix;
+import org.netbeans.spi.java.hints.TriggerTreeKind;
+import org.openide.util.NbBundle.Messages;
+
+@Hint(displayName = "#DN_ConvertTextBlockToString", description = "#DESC_ConvertTextBlockToString", category = "general", minSourceVersion = "13")
+@Messages({
+    "DN_ConvertTextBlockToString=Convert Text block to String",
+    "DESC_ConvertTextBlockToString=Text Block may not be supported in older versions of java older then 13"
+})
+public class ConvertTextBlockToString {
+
+    @TriggerTreeKind(Tree.Kind.STRING_LITERAL)
+    @Messages("ERR_ConvertTextBlockToString=Text block may not be supported")//NOI18N
+    public static ErrorDescription computeWarning(HintContext ctx) {
+        TokenSequence<?> ts = ctx.getInfo().getTokenHierarchy().tokenSequence();
```

Comment:
Maybe, should rock the `Document` when `TokenSequence` is got.
(The following log should be shown: !!WARNING!! Missing WRITE-LOCK when accessing TokenHierarchy)
@jlahoda Need not do that? 

Then should check whether `ts` is `null`.

Reply:
Can you please also tell that how to lock the document??

Reply:
e.g. see https://bits.netbeans.org/dev/javadoc/org-netbeans-modules-lexer/architecture-summary.html

@sdedic @jlahoda Could you tell @Akshay-Gupta-Oracle to handle it if you know the proper way in Java editor? or need not do that? Thanks.

Reply:
@sdedic Thank you for making it clear!

Reply:
Thank you @sdedic and @junichi11 for helping :)